### PR TITLE
Correct authentication and token generation v3 api

### PR DESCRIPTION
**Pull request by zhiyanliu**
1. Remove a wrong route mapper for '/v3/tokens' path.
2. Correct NOAUTH list to include path of authentication and token
   generation v3 api.
3. Correct authentication implementation of the api.

### CODE REVIEW

File: jumpgate/common/sl/auth.py [hunk 1]
```diff
@@ -54,11 +54,15 @@ def get_new_token_v3(credentials):
...
-                            'username')
+                            'identity',
+                            'password',
+                            'user',
+                            'name')
     credential = utils.lookup(credentials,
                               'auth',
-                              'passwordCredentials',
+                              'identity',
+                              'password',
```
Comment (jimlindeman) [inline jumpgate/common/sl/auth.py:64]: I see 'password' twice here.

Comment (zhiyanliu) [inline jumpgate/common/sl/auth.py:64]: Yep, pls refer https://github.com/openstack/python-keystoneclient/blob/master/keystoneclient/auth/identity/v3.py#L236

Comment (jimlindeman) [inline jumpgate/common/sl/auth.py:64]: Ah, okay so double nested "password" in the hash chain.

Comment (zhiyanliu) [inline jumpgate/common/sl/auth.py:64]: Correct. Could you pls help merging? =)

**Comment by sudorandom** [comment]
+1

**Comment by jimlindeman** [comment]
Sorry, been pulled away for other projects for a while.  Lets get this merged.

**Pull request merged**